pkgs/by-name/im/immich/package.nix

```diff
@@ -0,0 +1,241 @@
+{ lib
+, pkgs
+, buildNpmPackage
+, fetchFromGitHub
+, python3
+, nodejs
+, runCommand
+# build-time deps
+, pkg-config
+, makeWrapper
+, cmake
+, crane
+# runtime deps
+, ffmpeg
+, imagemagick
+, libraw
+, vips
+, perl
+}:
+let
+  buildNpmPackage' = buildNpmPackage.override { inherit nodejs; };
+  sources = lib.importJSON ./sources.json;
+  inherit (sources) version;
+
+  meta = with lib; {
+    description = "Self-hosted photo and video backup solution";
+    homepage = "https://immich.app/";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ jvanbruegge ];
+    inherit (nodejs.meta) platforms;
+  };
+
+  # The geodata website is not versioned,
+  # so we have to extract it from the container
+  geodata = runCommand "immich-geodata" {
+    outputHash = sources.geodata_hash;
+    outputHashMode = "recursive";
+  } ''
+    mkdir $out
+    echo "Downloading immich container image"
+    ${crane}/bin/crane export ghcr.io/immich-app/base-server-prod:${sources.container_tag} - \
```

Comment:
The comment mentions that this is done because the geonames data is not versioned, but upstream we're not relying on any particular version of it, so I'd recommend just getting it straight from geonames instead.

Reply:
The comment means that the geonames is not versioned in any way. Downloading it today is different from downloading it tomorrow

Reply:
Where does this geodata originate from?

Reply:
Found it, it's just three files fetched/upacked from somewhere else:

https://github.com/immich-app/base-images/blob/fe9bcfdbc28bb8a1f656cb924969428180269e4d/server/Dockerfile#L58-L60

It's like a dozen MiB total. We won't fetch a gigantic docker image just to extract these few files.

Reply:
We will have to find a stable mirror for these.

If push comes to shove, the internet archive could be used here as it shouldn't receive many requests for this in total (hydra.nixos.org + maybe a handful of independent builders).

Please also note that these have a different license to Immich (CC-BY-4.0) and that needs to be reflected in the drv.

Reply:
Upstream has agreed to publish a zip file with the geonames as part of their release workflow: https://github.com/immich-app/immich/pull/10916

Reply:
Hm, I'm a bit confused why you'd make the Immich project mirror those files? It'd have been sufficient to fetch a stable snapshot via the IA.

Reply:
They already mirror them in a sense via their github container repo. They even have a cronjob for that purpose. With the PR they just also attach them to the release without the rest of the container

Reply:
If they agreed to do so, I don't see the issue. Besides, probably better to hammer gh than ia.